pkg/prometheus/statefulset.go

```diff
@@ -310,23 +304,28 @@ func makeStatefulSetSpec(
 	shard int32,
 	ruleConfigMapNames []string,
 	tlsAssetSecrets []string,
-	version semver.Version,
 ) (*appsv1.StatefulSetSpec, error) {
 	// Prometheus may take quite long to shut down to checkpoint existing data.
 	// Allow up to 10 minutes for clean termination.
 	terminationGracePeriod := int64(600)
 
-	prometheusImagePath, err := operator.BuildImagePath(
+	pVersion := operator.StringValOrDefault(p.Spec.Version, operator.DefaultPrometheusVersion)
+	pImagePath, err := operator.BuildImagePath(
 		operator.StringPtrValOrDefault(p.Spec.Image, ""),
 		operator.StringValOrDefault(p.Spec.BaseImage, c.PrometheusDefaultBaseImage),
-		p.Spec.Version,
-		p.Spec.Tag,
-		p.Spec.SHA,
+		pVersion,
+		operator.StringValOrDefault(p.Spec.Tag, ""),
```

Comment:
Do we need `operator.StringValOrDefault()` here since default string value is `""`?

Reply:
Only trying to keep the code consistent since `alertmanager/statefulset.go` also implements the same behaviour. Will be happy to revert this, but if we do so I would be in favor of also removing this function call in the `alertmanager/statefulset.go` code

[1] https://github.com/prometheus-operator/prometheus-operator/blob/4e026ff8f9ccac7ab12a7118a94f316ae5e8f183/pkg/alertmanager/statefulset.go#L226-L227

Reply:
A follow-up PR to clean up everything sounds good to me :)